test/interfaceTests.jl

```diff
@@ -259,15 +260,42 @@ end
-        @test_skip 1==0#toDotFile(dfg, "something.dot")
-    else
-        @test toDot(dfg) == "graph graphname {\n18 [\"label\"=\"x8x9f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n2 [\"label\"=\"x2\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n2 -- 11\n2 -- 12\n16 [\"label\"=\"x6x7f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n11 [\"label\"=\"x1x2f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n7 [\"label\"=\"x7\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n7 -- 16\n7 -- 17\n9 [\"label\"=\"x9\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n9 -- 18\n9 -- 19\n10 [\"label\"=\"x10\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n10 -- 19\n19 [\"label\"=\"x9x10f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n17 [\"label\"=\"x7x8f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n8 [\"label\"=\"x8\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n8 -- 17\n8 -- 18\n6 [\"label\"=\"x6\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n6 -- 15\n6 -- 16\n4 [\"label\"=\"x4\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n4 -- 13\n4 -- 14\n3 [\"label\"=\"x3\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n3 -- 12\n3 -- 13\n5 [\"label\"=\"x5\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n5 -- 14\n5 -- 15\n13 [\"label\"=\"x3x4f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n14 [\"label\"=\"x4x5f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n15 [\"label\"=\"x5x6f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n12 [\"label\"=\"x2x3f1\",\"shape\"=\"ellipse\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n1 [\"label\"=\"x1\",\"shape\"=\"box\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n1 -- 11\n}\n"
+@testset "Summaries and Summary Graphs" begin
+    factorFields = fieldnames(DFGFactorSummary)
+    variableFields = fieldnames(DFGVariableSummary)
 
-        @test toDotFile(dfg, "something.dot") == nothing
-        Base.rm("something.dot")
+    summary = getSummary(dfg)
+    @test symdiff(collect(keys(summary.variables)), ls(dfg)) == Symbol[]
+    @test symdiff(collect(keys(summary.factors)), lsf(dfg)) == Symbol[]
+
+    summaryGraph = getSummaryGraph(dfg)
+    @test symdiff(ls(summaryGraph), ls(dfg)) == Symbol[]
+    @test symdiff(lsf(summaryGraph), lsf(dfg)) == Symbol[]
+    # Check all fields are equal for all variables
+    for v in ls(summaryGraph)
+        for field in variableFields
+            @test getfield(getVariable(dfg, v), field) == getfield(getVariable(summaryGraph, v), field)
+        end
     end
+    for f in lsf(summaryGraph)
+        for field in factorFields
+            @test getfield(getFactor(dfg, f), field) == getfield(getFactor(summaryGraph, f), field)
+        end
+    end
+end
+
+@testset "Producing Dot Files" begin
```

Comment:
@GearsAD Please have a look if you are ok with this simpler graph for testing toDot

Reply:
Cool, LGTM!